ecdsa/ta/ecdsa_ta.c

```diff
@@ -0,0 +1,342 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+ */
+
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <ecdsa_ta.h>
+
+struct ecdsa {
+	TEE_ObjectHandle keypair;
+	TEE_ObjectHandle public_key;
+	uint32_t algo_id;
+	uint32_t hash_algo;
+	uint32_t key_sz;
+};
+
+TEE_Result TA_CreateEntryPoint(void)
+{
+	DMSG("TA created");
+	return TEE_SUCCESS;
+}
+
+void TA_DestroyEntryPoint(void)
+{
+	/* TA destroyed */
+}
+
+TEE_Result TA_OpenSessionEntryPoint(uint32_t param_types,
+				    TEE_Param params[4],
+				    void **sess_ctx)
+{
+	struct ecdsa *sess;
+
+	sess = TEE_Malloc(sizeof(*sess), 0);
+	if (!sess)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	sess->keypair = TEE_HANDLE_NULL;
+	sess->public_key = TEE_HANDLE_NULL;
+
+	(void)param_types;
```

Comment:
Please use the `__unused` attribute on `param_types` above instead, same for `params`.